legacy/builder/builder_utils/utils.go

```diff
@@ -153,17 +67,35 @@ func findAllFilesInFolder(sourcePath string, recurse bool) ([]string, error) {
 	return sources, nil
 }
 
-func compileFilesWithRecipe(ctx *types.Context, sourcePath *paths.Path, sources paths.PathList, buildPath *paths.Path, buildProperties *properties.Map, includes []string, recipe string) (paths.PathList, error) {
+func CompileFiles(ctx *types.Context, sourcePath *paths.Path, recurse bool, buildPath *paths.Path, buildProperties *properties.Map, includes []string) (paths.PathList, error) {
```

Comment:
I think it would clearer to make `CompileFiles` private without changing its signature and instead have two public functions `CompileFiles` and `CompileFilesRecursive` or similar naming but without the `recurse` argument.

So something like this:
```
builder_utils.CompileFiles(ctx, variantFolder, true, buildPath, buildProperties, includes)
```
can be written like this:
```
builder_utils.CompileFilesRecursive(ctx, variantFolder, buildPath, buildProperties, includes)
```

Another example:
```
builder_utils.CompileFiles(ctx, library.SourceDir, false, libraryBuildPath, buildProperties, includes)
```
becomes:
```
builder_utils.CompileFiles(ctx, library.SourceDir, libraryBuildPath, buildProperties, includes)
```


Since we'll need to touch the `legacy` package more and more I think this is a valuable change that will make it easier to understand certain code snippets.